test/integration-test/TestIIRDeserializer.cpp

```diff
@@ -0,0 +1,607 @@
+    }
+    stage.update(iir::NodeUpdateType::level);
+  }
+  for(const auto& MSPtr : iterateIIROver<iir::Stage>(*(target->getIIR()))) {
+    MSPtr->update(iir::NodeUpdateType::levelAndTreeAbove);
+  }
+}
+
+static void readIIRFromFile(OptimizerContext &optimizer, std::shared_ptr<iir::StencilInstantiation>& target, std::string fname) {
+  target =
+      IIRSerializer::deserialize(fname, &optimizer, IIRSerializer::SK_Json);
+
+  //this is whats actually to be tested.
+  optimizer.restoreIIR("<restored>", target);
+}
+
+static bool compareIIRs(std::shared_ptr<iir::StencilInstantiation> lhs, std::shared_ptr<iir::StencilInstantiation> rhs) {
+  //first compare the (structure of the) iirs, this is a precondition before we can actually check the metadata / derived info
+  IIR_EARLY_EXIT((compareIIRstructures(lhs->getIIR().get(), rhs->getIIR().get())));
+
+  //then we compare the meta data
+  IIR_EARLY_EXIT((compareMetaData(lhs->getMetaData(), rhs->getMetaData())));
+
+  //and finally the derived info 
+  IIR_EARLY_EXIT((compareMetaData(lhs->getMetaData(), rhs->getMetaData())));
+  ((compareDerivedInformation(lhs->getIIR().get(), rhs->getIIR().get())));
+
+  return true;
+}
+
+TEST(IIRDeserializerTest, CopyStencil) {
+  char cCurrentPath[FILENAME_MAX];
+  getcwd(cCurrentPath, sizeof(cCurrentPath));
+  cCurrentPath[sizeof(cCurrentPath) - 1] = '\0'; /* not really required */
+  printf ("The current working directory is %s", cCurrentPath);
+
+  Options compileOptions;
+  OptimizerContext::OptimizerContextOptions optimizerOptions;
+  DawnCompiler compiler(&compileOptions);   
+  OptimizerContext optimizer(compiler.getDiagnostics(), optimizerOptions, std::make_shared<dawn::SIR>());
+
+  //read IIRs from file
+  std::shared_ptr<iir::StencilInstantiation> copy_stencil_from_file = std::make_shared<iir::StencilInstantiation>(
+    *optimizer.getSIR()->GlobalVariableMap, optimizer.getSIR()->StencilFunctions);
+  readIIRFromFile(optimizer, copy_stencil_from_file, "reference_iir/copy_stencil.iir");
+  UIDGenerator::getInstance()->reset();
+
+  //generate IIRs in memory
+  std::shared_ptr<iir::StencilInstantiation> copy_stencil_memory = std::make_shared<iir::StencilInstantiation>(
+    *optimizer.getSIR()->GlobalVariableMap, optimizer.getSIR()->StencilFunctions);
+  createCopyStencilIIRInMemory(copy_stencil_memory);
+  UIDGenerator::getInstance()->reset(); 
+
+  EXPECT_TRUE(compareIIRs(copy_stencil_from_file, copy_stencil_memory));
+}
+
+TEST(IIRDeserializerTest, LaplStencil) {
+  Options compileOptions;
+  OptimizerContext::OptimizerContextOptions optimizerOptions;
+  DawnCompiler compiler(&compileOptions);   
+  OptimizerContext optimizer(compiler.getDiagnostics(), optimizerOptions, std::make_shared<dawn::SIR>());
+
+  //read IIRs from file
+  std::shared_ptr<iir::StencilInstantiation> lap_stencil_from_file = std::make_shared<iir::StencilInstantiation>(
+    *optimizer.getSIR()->GlobalVariableMap, optimizer.getSIR()->StencilFunctions);
```

Comment:
As far as I remember you can just pass empty maps here. No need to use OptimzierContext.